cmd/jx/main.go

```diff
@@ -5,34 +5,42 @@ import (
 	"os"
 
 	"foxygo.at/jsonnext"
+	jxkong "foxygo.at/jsonnext/kong"
 	"github.com/alecthomas/kong"
 	"github.com/google/go-jsonnet"
 )
 
-var CLI struct {
-	ImportPath []string `name:"jpath" sep:":" short:"J"`
-	Filename   string   `arg:"" optional:"" help:"File to evaluate. stdin is used if omitted or \"-\""`
+type config struct {
+	jxkong.Config
+	Filename string `arg:"" optional:"" help:"File to evaluate. stdin is used if omitted or \"-\""`
 }
 
 func main() {
-	kong.Parse(&CLI)
+	cli := parseCLI()
 
-	vm := jsonnet.MakeVM()
-	i := newImporter(CLI.ImportPath)
-	vm.Importer(i)
+	vm := newVM(cli.Config.Config)
 
-	out, err := run(vm, CLI.Filename)
+	out, err := run(vm, cli.Filename)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 	fmt.Print(out)
 }
 
-func newImporter(importPath []string) *jsonnext.Importer {
-	i := jsonnext.Importer{SearchPath: importPath}
-	i.AppendSearchFromEnv("JXPATH")
-	return &i
+func parseCLI() *config {
+	c := &config{Config: *jxkong.NewConfig()}
+	kong.Parse(c)
+	return c
+}
+
+func newVM(c *jsonnext.Config) *jsonnet.VM {
```

Comment:
I wonder - and I really not more than just not knowing:
could you have a `jsonnext.Config.MakeVM()` ?

Reply:
I could have `MakeVM(pathEnvVar string)` - i think that makes sense since it wraps up all config from this package. One may not want to use the Importer, but then one can configure the bits individually you want. Good call. Will do.

Reply:
Done.